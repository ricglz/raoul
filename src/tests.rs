use super::{parse, DirFunc};

#[test]
fn use_of_undeclared_variable() {
    let filename = "examples/invalid/undeclared-variable.ra";
    let program = std::fs::read_to_string(filename).expect(filename);
    let debug = true;
    let ast_response = parse(&program, debug);
    assert!(ast_response.is_ok());
    let ast = ast_response.unwrap();
    let mut dir_func = DirFunc::new();
    let build_res = dir_func.build_dir_func(ast);
    assert!(build_res.is_err());
}
