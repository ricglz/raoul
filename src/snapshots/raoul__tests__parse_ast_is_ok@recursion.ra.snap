---
source: src/tests.rs
expression: quad_manager
input_file: src/examples/invalid/dynamic/recursion.ra
---
0 - Goto       -     -     4    
1 - Era        0     1     -    
2 - GoSub      1     -     -    
3 - EndProc    -     -     -    
4 - Era        0     1     -    
5 - GoSub      1     -     -    
6 - End        -     -     -    

